### DISCUSSION TITLE
How to load balanace two or more FTP servers?

### DISCUSSION THREAD

**Discussion by mezuky04** [drakkan/sftpgo#1065] [General]
Hello. I tried to load balance two FTP servers using TLS with HAProxy following [this tutorial](https://web.archive.org/web/20130404121626/http://ben.timby.com/?page_id=210) but I'm not able to make it work. When I try to connect with Filezilla, I get the following error:

<img width="594" alt="image" src="https://user-images.githubusercontent.com/7033003/203083701-613db6dd-c98b-47f8-8b6c-8818d7bcac61.png">

Could you make a guide about how to load balance FTP servers? 🙂

**Comment by drakkan** [comment]
You client is trying to open a passive connection to the IP 35.181.144.21, port 50036 and it gets a connection refused error. You have to allow and forward the data connection ports and configure your proxy to send the proxy header for both control and data connections. I suggest to study how FTP works

**Reply by mezuky04** [reply]
The problem was that I configured different port ranges in HAProxy which didn't match the default `passive_port_range`. Thanks for your reply 🙂

**Reply by drakkan** [reply]
> The problem was that I configured different port ranges in HAProxy which didn't match the default `passive_port_range`. Thanks for your reply slightly_smiling_face

Thanks for reporting back, glad it works now

**Comment by XPOL555** [comment]
hi @mezuky04 can you help me with this https://github.com/drakkan/sftpgo/issues/1067?
